src/log/syslog.rs

```diff
@@ -19,7 +19,32 @@ impl Log for Syslog {
         };
 
         let message = format!("{}", record.args());
-        syslog(priority, libc::LOG_AUTH, &message);
+        let message_len = message.bytes().len();
+
+        let mut x: usize = 960;
+        let mut x_prev: usize = 0;
+
+        loop {
+            // floor_char_boundary is currently unstable
+            while !message.is_char_boundary(x) {
+                x -= 1;
+            }
+
+            if x < message_len {
+                x = message[x_prev..x].rfind(char::is_whitespace).unwrap_or(x) + x_prev + 1;
+            } else {
+                x = message_len
+            }
+
+            syslog(priority, libc::LOG_AUTH, &message[x_prev..x]);
+
+            x_prev = x;
+            x += 960;
+
+            if x_prev >= message_len {
```

Comment:
If this is the break condition and `x_prev` just got assigned `x`. Wouldn't it be possible to reorganize this `loop` so it becomes something like `while x >= message_len { ... }`?

Reply:
This was necessary  because the loop needed to run at least once, and decide to continue at the end.
However with your first suggestion this is no longer necessary.